### PR TITLE
Feat: aineko dream CLI

### PR DESCRIPTION
**Pull request by craycrayfish**
## Summary

The PR adds an `aineko dream` CLI command that queries the Aineko Dream API to generate a pipeline. Changes are made on top of #104 , merge that first.

To use, 
```
poetry run aineko dream "prompt here" "API_KEY_HERE" --url https://dream-dev.ap-2359264f-9cba-43ba-9ac4-ae3ba7b1d4b1.aineko.app
```

optionally, the `--url` option to use is used to specify the dev version of the Aineko Dream API. It defaults to the production version when nothing is specified.

## Motivation

The Aineko Dream API should be integrated to Aineko CLI.

## Author Checklist

- [x] Changes are motivated and described in the PR description.
- [ ] Testing strategy is described if automated tests are not included in the PR.
- [x] Documentation is updated if necessary.
- [x] Status checks pass.
- [ ] If the version number is changed, it is updated in `pyproject.toml` , `aineko/__init__.py`, `aineko/templates/first_aineko_pipeline/{{cookiecutter.project_slug}}/pyproject.toml`.

## Reviewer Checklist

- [ ] Title is accurate and formatted appropriately.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Docs are updated if necessary.
- [ ] Code is pythonic and consistent with the rest of the codebase.
- [ ] The code is consistent with the [style guide](https://google.github.io/styleguide/pyguide.html).

### CODE REVIEW

**Comment by Barabazs** [comment]
> Changes are made on top of #104 , merge that first.

Then you should branch of `feat/create-from-repo` to make your incremental changes and create a PR to merge it into `feat/create-from-repo`. Otherwise we need to review the same code twice, which I'm not going to do when there are this many changes.

File: aineko/cli/dream.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+@click.argument("prompt")
+@click.argument("api-key")
+@click.option(
+    "-u",
+    "--url",
+    default="https://dream.ap-2359264f-9cba-43ba-9ac4-ae3ba7b1d4b1.aineko.app",
+    help="API url to use for the Aineko Dream API.",
+)
+def dream(prompt: str, api_key: str, url: str) -> None:
+    """Main function generate an aineko project using Aineko Dream.\f
```
Comment (mdigi14) [inline aineko/cli/dream.py]:
```suggestion
    """Main function to generate an aineko project using Aineko Dream.\f
```

File: aineko/cli/dream.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        url: URL to use for the Aineko Dream API.
+    """
+    try:
+        click.echo("Generating project from prompt...")
+
+        r = requests.get(
+            url=f"{url}/create/",
+            params={"prompt": prompt},
+            headers={"x-api-key": api_key},
+            timeout=300,
```
Comment (mdigi14) [inline aineko/cli/dream.py:33]: Might want to make this a config, also is this set to 300 for any reason? Is this based on the timeout for GPT-4?

File: docs/aineko_dream.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Aineko Dream
+
+Aineko Dream is the quickest way to get started with Aineko pipelines by leveraging the power of generative AI to create a starter Aineko pipeline based on your use case.
```
Comment (mdigi14) [inline docs/aineko_dream.md]: I would make this more concise and use the active voice. Using Aineko 3 times sounds too repetitive imo.

```suggestion
Aineko Dream leverages the power of generative AI to create a starter Aineko pipeline based on your use case. 
```

File: docs/aineko_dream.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## Generating a Project
+
+To generate a project, invoke the Aineko Dream CLI with a prompt with:
+
+```bash
+poetry run aineko dream "create a pipeline that scrapes twitter and analyses the results to identify trends" "API-KEY"
+```
+
+replacing `API-KEY` with a valid Aineko Dream API key. Contact support@aineko.dev to get an API key to try this feature.
```
Comment (mdigi14) [inline docs/aineko_dream.md:13]: Linking to Slack here also might be good. A general response time is also helpful (ie expect a response within 72 hours)

File: aineko/cli/dream.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import requests
+
+
+@click.command()
+@click.argument("prompt")
+@click.argument("api-key")
+@click.option(
+    "-u",
+    "--url",
+    default="https://dream.ap-2359264f-9cba-43ba-9ac4-ae3ba7b1d4b1.aineko.app",
```
Comment (mdigi14) [inline aineko/cli/dream.py:15]: Should we make this a config? As we continue to shuffle some backend infrastructure this Project ID might change.

**Review by mdigi14 — APPROVED** [review]
Nice feature. I'm approving but I left some minor suggestions I think we should incorporate. 👍

File: aineko/cli/dream.py (hunk 2 above)
Comment (craycrayfish) [reply aineko/cli/dream.py:33]: timeout is based on our load balancer timeouts

**Pull request merged**